File: src/main/java/de/twometer/amongus/gui/CreditsPage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.core.AmongUs;
import de.twometer.amongus.model.GameState;
import de.twometer.neko.Neko;
import de.twometer.neko.sound.SoundSource;

public class CreditsPage extends BasePage {

    private SoundSource soundSource;

    public CreditsPage() {
        super("Credits.html");
    }

    @Override
    public void onUnload() {
        super.onUnload();
        soundSource.stop();
        amongUs.getStateController().changeState(GameState.Menus);
    }

    @Override
    public void onDomReady() {
        soundSource = AmongUs.get().getSoundFX().play("CreditsRoll.ogg");
        context.setElementText("nekoVersion", Neko.VERSION);
        amongUs.getStateController().changeState(GameState.Credits);
    }

    public void back() {
        goBack();
    }

}

File: src/main/kotlin/de/twometer/amogus/gui/CreditsPage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.StateManager
import de.twometer.amogus.model.GameState
import de.twometer.neko.Neko
import de.twometer.neko.audio.SoundEngine
import de.twometer.neko.audio.SoundSource

class CreditsPage : BasePage("Credits.html") {

    private lateinit var soundSource: SoundSource

    override fun onLoaded() {
        setElementText("nekoVersion", Neko.VERSION)
        StateManager.changeGameState(GameState.Credits)
        soundSource = SoundEngine.play("CreditsRoll.ogg")
    }

    override fun onUnloaded() {
        soundSource.stop()
        StateManager.changeGameState(GameState.Menus)
    }
}